src/engine.c

```diff
@@ -270,6 +276,26 @@ ucschar_strlen (const ucschar* str)
     return p - str;
 }
 
+static gboolean
+check_client_commit (const gchar *version)
+{
+    gboolean client_commit = FALSE;
+    gint len = 0;
+
+    gchar **versions = g_strsplit_set (version, ".", 3);
+    len = g_strv_length (versions);
+    if (len != 3)
+        return client_commit;
+
+    client_commit = atoi (versions[0]) >= 1 &&
+        atoi (versions[1]) >= 5 &&
+        atoi (versions[2]) >= 20;
```

Comment:
The version comparision should be like `(major > 1) || ((major == 1) && ((minor > 5) || ((minor == 5) && (micro >= 20))))`.

Reply:
I suggest using IBUS_CHECK_VERSION() macro here.

Reply:
Done